src/System.Net.Http/src/System/Net/Http/Headers/HeaderUtilities.cs

```diff
@@ -300,6 +300,11 @@ internal static void CheckValidQuotedString(string value, string parameterName)
             {
                 return (DateTimeOffset)storedValue;
             }
+            else if (defaultValue != null && store.Contains(descriptor))
```

Comment:
What's the purpose of the `&& store.Contains(descriptor)`?  I'm wondering why this can't just be:
```C#
return storedValue != null ? (DateTimeOffset)storedValue : defaultValue;
```

Reply:
I did something similar but that did not work as I wanted because that returns default even in case Expires header does not exist.

Reply:
I see. Ok.

There's some inefficiency here, in that `store.GetParsedValues` calls `TryGetAndParseHeaderInfo` and then `store.Contains` calls it again; the only difference between the two calls is that the former returns details from the resulting info whereas Contains just returns whether it got one or not.

I've not looked to see if this is on a hot path.  If it's not, this seems fine.  If it is, we should refactor things or expose an additional internal method to allow performing the lookup only once.

Reply:
I'll can take another look. This is why I check defaultValue first so this path is currently hit only for Expires when the value is incorrect or missing. 
Since we do not really care about expiration as HttpClient level, this should not be accessed and I did not find any reference to it in production code.